test/reduce/reduce_no_index.cpp

```diff
@@ -1,248 +1,148 @@
-                                                              reduceDims,
-                                                              reduceOpId,
-                                                              propagateNan,
-                                                              false,
-                                                              alpha,
-                                                              beta);
-    }
-    else if(data_type == 5)
-    {
-        result = profile_reduce_impl<ck::bhalf_t, float, ck::bhalf_t>(true,
-                                                                      init_method,
-                                                                      false,
-                                                                      false,
-                                                                      inLengths,
-                                                                      reduceDims,
-                                                                      reduceOpId,
-                                                                      propagateNan,
-                                                                      false,
-                                                                      alpha,
-                                                                      beta);
-    }
-    else if(data_type == 6)
-    {
-        result = profile_reduce_impl<double, double, double>(true,
-                                                             init_method,
-                                                             false,
-                                                             false,
-                                                             inLengths,
-                                                             reduceDims,
-                                                             reduceOpId,
-                                                             propagateNan,
-                                                             false,
-                                                             alpha,
-                                                             beta);
-    }
+template <typename TType>
+class ReduceWithIndexHalf : public ReduceWithIndexTest<TType>
+{
+};
 
-    return (result);
+template <typename TType>
+class ReduceWithIndexBHalfFloat : public ReduceWithIndexTest<TType>
+{
 };
 
-constexpr ReduceTensorOp reduceOpId = ReduceTensorOp::AVG;
-constexpr bool propagateNan         = false;
+TYPED_TEST_SUITE(ReduceWithIndexFloat, Reduce_float_types);
+TYPED_TEST_SUITE(ReduceWithIndexDouble, Reduce_double_types);
+TYPED_TEST_SUITE(ReduceWithIndexInt8, Reduce_int8t_types);
+TYPED_TEST_SUITE(ReduceWithIndexHalf, Reduce_half_types);
+TYPED_TEST_SUITE(ReduceWithIndexBHalfFloat, Reduce_bhalf_float_Types);
 
-int main(int argc, char* argv[])
+TYPED_TEST(ReduceWithIndexFloat, ReduceWithIndexTestFloat)
```

Comment:
Can we have here separate test cases for each reduce op? If any of them will fail we would get clear result indicating what exactly failed.